From 0e24164342274d86e00f4c18f5518faeafb297fa Mon Sep 17 00:00:00 2001
From: Timo Wendt <timo@tjwendt.de>
Date: Tue, 25 Jun 2013 23:15:44 +0200
Subject: [PATCH] Irda: Added IrDA HAL Library

PatchSet 2: Cleanup

PatchSet 3: Added some more documentaion

Patchset 4: Removed references to Galaxy S4 in the commit message.  This HAL
is supposed to be generic in the long term, and at the very least, works
on other Samsung devicessuch as n80xx.

Change-Id: I8af12b8716791bb13ba7b85b9cab8d2a1c5fed18
---
 include/hardware/irda.h | 45 +++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 45 insertions(+)
 create mode 100644 include/hardware/irda.h

diff --git a/include/hardware/irda.h b/include/hardware/irda.h
new file mode 100644
index 0000000..edd30df
--- /dev/null
+++ b/include/hardware/irda.h
@@ -0,0 +1,45 @@
+/*
+ * Copyright (C) 2013 CyanogenMod Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef ANDROID_IRDA_INTERFACE_H
+#define ANDROID_IRDA_INTERFACE_H
+
+#include <stdint.h>
+#include <sys/cdefs.h>
+#include <sys/types.h>
+
+#include <hardware/hardware.h>
+
+__BEGIN_DECLS
+
+#define IRDA_HARDWARE_MODULE_ID "irda"
+
+struct irda_device_t {
+    struct hw_device_t common;
+
+    /*
+     * (*sendIrCode)() Send IR code
+     *
+     * @param ircode is the code to send
+     * @param length is the length of the code string
+     *
+     */
+    void (*send_ircode)(char* ircode, int length);
+};
+
+__END_DECLS
+
+#endif // ANDROID_IRDA_INTERFACE_H
-- 
1.8.3.2

